app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,58 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    edition.temporary_first_published_at = previous_live_version&.first_published_at
+    edition.temporary_last_edited_at = now
+    edition.publisher_first_published_at = payload[:first_published_at]
+
+    unless edition.update_type == "major"
+      edition.major_published_at = previous_live_version&.major_published_at
+    end
+
+    edition.first_published_at = payload[:first_published_at] ||
+      previous_live_version&.first_published_at
+
+    edition.last_edited_at = payload[:last_edited_at] || now
+
+    edition.public_updated_at = payload[:public_updated_at] ||
+      previous_live_version&.public_updated_at
+
+    edition.publisher_major_published_at = payload[:public_updated_at]
+    edition.publisher_published_at = nil
+    edition.publisher_last_edited_at = payload[:last_edited_at]
+
+    edition.save!
+  end
+
+  def self.live_transition(
+    edition,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    unless edition.temporary_first_published_at.present?
+      edition.temporary_first_published_at = now
+    end
+
+    if edition.update_type == "major"
+      edition.major_published_at = now
+    else
+      edition.major_published_at = previous_live_version&.major_published_at
```

Comment:
Might be worth adding a comment or something here to explain that the else condition of this can be removed once update_type is no longer part of publish.